spharpy/spherical.py

```diff
@@ -282,35 +302,41 @@ def spherical_harmonic_basis(n_max, coords):
 
     Examples
     --------
-
     >>> import spharpy
     >>> n_max = 2
-    >>> coords = spharpy.samplings.icosahedron()
-    >>> Y = spharpy.spherical.spherical_harmonic_basis(n_max, coords)
+    >>> coordinates = spharpy.samplings.icosahedron()
+    >>> Y = spharpy.spherical.spherical_harmonic_basis(n_max, coordinates)
 
     """
 
-    n_coeff = (n_max+1)**2
+    n_coeff = (n_max + 1) ** 2
 
-    basis = np.zeros((coords.csize, n_coeff), dtype=complex)
+    basis = np.zeros((coordinates.csize, n_coeff), dtype=complex)
 
     for acn in range(n_coeff):
-        order, degree = acn_to_nm(acn)
+        if channel_convention == "fuma":
+            order, degree = fuma_to_nm(acn)
+        else:
+            order, degree = acn_to_nm(acn)
         basis[:, acn] = _special.spherical_harmonic(
-            order,
-            degree,
-            coords.colatitude,
-            coords.azimuth)
-
+            order, degree, coordinates.colatitude, coordinates.azimuth
+        )
+        if normalization == "sn3d":
+            basis[:, acn] *= n3d_to_sn3d_norm(order)
+        elif normalization == "maxN":
+            basis[:, acn] *= n3d_to_maxn(acn)
+        if phase_convention is None:
+            # Condon-Shortley phase term is already included in
+            # the special.spherical_harmonic function
+            # so need to divide by (-1)^m
+            basis[:, acn] /= (-1) ** float(degree)
     return basis
 
 
-def spherical_harmonic_basis_gradient(n_max, coords):
+def spherical_harmonic_basis_gradient(n_max, coordinates):
```

Comment:
You did not implement the phase and normalization conventions for the gradients. Is there a reason for this?
Did you want to keep this for a separate PR?